### PR TITLE
Add a new PR to merge in dev as requested by reviewer

### PR DESCRIPTION
**Pull request by altontonn**
# Budget app 💎 

- Splash screen

     - A simple page with the name of your app (yes, you need to choose one), and links to the sign-up and log-in pages.

- Sign up and login pages

      - The user should be able to register in the app with full name, email, and password (all mandatory).
      - The user can log into the app using email and password.
      - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)

     - When the user logs in, they are presented with the categories page.
     - For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.
     - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
     - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page

     - For a given category, the list of transactions is presented, ordered by the most recent.
     - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
     - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
     - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page

       - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
       - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
       - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page

      - The user fills out a form to create a new transaction with:
        - name (mandatory)
        - amount (mandatory)
        - categories (mandatory at least one)
      - The user clicks (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
      - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

- Testing requirements

     - Create unit and integration tests for all the most important components of your RoR application

### CODE REVIEW

File: app/views/accounts/_form.html.erb [hunk 1]
```diff
@@ -4,11 +4,9 @@
     <%= form.label "Name", style: "display: block" %>
     <%= form.text_field :name, required: true, class: "form-control"%>
   </div>
-
   <div class="form-group">
     <%= form.label "Amount", style: "display: block" %>
     <%= form.number_field :amount, required: true, class: "form-control" %>
   </div>
-
   <%= form.submit "Create Transaction", class: "btn btn-primary mt-4"%>
```
Comment (addod19) [inline app/views/accounts/_form.html.erb]: - [ ] In the `Add a new transaction` page, we are required to fill out 3 mandatory inputs as we can see in the project requirements below. However, we are missing the `categories` input field
Screenshot from 2023-04-24 22-10-43

**Review by addod19 — CHANGES_REQUESTED** [review]
### Hi @altontonn, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [ ] Correct Git-flow followed :100: 
- [ ] Registration and login functionality works as expected and created with Devise :rocket: 
- [ ] Homepage and transaction pages implemented :100: 
- [ ] App is deployed and available in the README file :fireworks: 
- [ ] Validation implemented :+1: 
- [ ] Postgres used as database :1st_place_medal: 


## Required Changes ♻️
_Check the comments under the review._
- [ ] It would be better if we write unit and integration tests for all the most important components of our RoR application. For integration testing with Rspec and Capybara, we can use `visit sign_up_path` to navigate to the registration page, then use `fill_in 'Email', with: email` to enter email credentials, for click action we can use `click_button 'Sign up'` to trigger the signup action
- [ ] It would be better if we fix the MIT license link, it is currently broken and navigate us to `404` error page not found
- [ ] In the video presentation, we are required to highlight a piece of code we are proud of, I did not see that part in the presentation. Kindly add it
Screenshot from 2023-04-24 22-38-01



## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/accounts/_form.html.erb (hunk 1 above)
Comment (rloterh) [reply app/views/accounts/_form.html.erb]: resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Newton,  

Your project is complete! 🏁 Great job! There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job implementing recommended changes!
🎲  Properly added required models, controllers, and corresponding views
🎲  Devise correctly implemented for authentication!
🎲  Schema follow given ERD!

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
[ ] In addition, kindly ensure that provided placeholder texts/guide instructions are removed and/or replaced with appropriate contents applicable under each section of your ReadMe. 

**Examples of such are the underlined:**
image

Implementing the above change(s) will make your ReadMe/ repository stand out even better! 😉

Keep soaring higher! 💻  🍷  🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**